src/services/ls-migration/addressBook.ts

```diff
@@ -26,3 +27,45 @@ export const migrateAddressBook = (lsData: LOCAL_STORAGE_DATA): AddressBookState
     }
   }
 }
+
+// Temporary post-migration fix for malformed data
+export const sanitizeMigratedAddressBook = (state: AddressBookState): AddressBookState => {
+  const hasValidAb = Object.values(state).every((ab) => {
+    return Object.keys(ab).every(utils.isAddress)
+  })
+
+  if (hasValidAb) {
+    return state
+  }
+
+  const sanitizedAb = Object.keys(state).reduce<AddressBookState>((sanitizedAb, chainId) => {
+    const chainAb = state[chainId]
+
+    const sanitizedChainAb = Object.keys(chainAb).reduce<AddressBook>((sanitizedChainAb, address) => {
+      if (!address || !utils.isAddress(address)) {
+        return sanitizedChainAb
+      }
+
+      const name = chainAb[address]
+      if (!name) {
```

Comment:
What about a name only consisting of whitespace?

Reply:
Good catch! We now `trim` this.